playbooks/roles/ocp-config/tasks/main.yaml

```diff
@@ -56,6 +56,26 @@
     args:
       chdir: "{{ workdir }}"
 
+  - name: Generate Machine Config Master
+    template:
+      src: ../templates/99-master-machineconfig.yaml.j2
+      dest: "{{ workdir }}/openshift/99-master-machineconfig.yaml"
+    when:
+      - luks_compliant.enabled
+      - luks_config | length > 0
+      - luks_config[0].thumbprint is not none
+      - luks_config[0].url is not none
+
+  - name: Generate Machine Config Worker
+    template:
+      src: ../templates/99-worker-machineconfig.yaml.j2
+      dest: "{{ workdir }}/openshift/99-worker-machineconfig.yaml"
+    when:
+      - luks_compliant.enabled
+      - luks_config | length > 0
+      - luks_config[0].thumbprint is not none
+      - luks_config[0].url is not none
+
```

Comment:
I see both master and worker configuration is same but just the file name difference: Can we reuse code and template file to loop for master, worker?

Reply:
As per our understanding, it is better to have two seperate configuration files as we need to place them at desired location during deployment, which will get removed after completion of task 
The core element in this file is the role either master or worker. So if we think about reusing the code then we need to add loop for the same which will create the file with the expected role and filename.
So if we have different two template files it will be feasible.

Reply:
@yussufsh Updated code as per your suggestion. I have added code which will reuse the  the same template file to create MachineConfig for master and worker.

Reply:
looks much better now! thanks.